topi/python/topi/cuda/depthwise_conv2d.py

```diff
@@ -186,3 +186,102 @@ def traverse(OP):
 
     traverse(outs[0].op)
     return s
+
+
+def schedule_depthwise_conv2d_back_input_nhwc(outs):
+    """Schedule for depthwise_conv2d nhwc backward wrt input.
+
+    Parameters
+    ----------
+    outs: Array of Tensor
+        The computation graph description of depthwise_conv2d
+        in the format of an array of tensors.
+
+    Returns
+    -------
+    s: Schedule
+        The computation schedule for depthwise_conv2d nhwc.
+    """
+    outs = [outs] if isinstance(outs, tvm.tensor.Tensor) else outs
+    s = tvm.create_schedule([x.op for x in outs])
+
+    def _schedule(Padded_out_grad, In_grad):
+        s[Padded_out_grad].compute_inline()
+
+        block_x = tvm.thread_axis("blockIdx.x")
+        thread_x = tvm.thread_axis("threadIdx.x")
+        _, h, w, c = In_grad.op.axis
+
+        fused_wc = s[In_grad].fuse(w, c)
+
+        fused_hwc = s[In_grad].fuse(h, fused_wc)
+        xoc, xic = s[In_grad].split(fused_hwc, factor=128)
+
+        s[In_grad].bind(xoc, block_x)
+        s[In_grad].bind(xic, thread_x)
+
+    def traverse(OP):
+        # inline all one-to-one-mapping operators except the last stage (output)
+        if OP.tag == 'depthwise_conv2d_back_input_nhwc':
+            Padded_out_grad = OP.input_tensors[0]
+            Dilate_out_grad = Padded_out_grad.op.input_tensors[0]
+            s[Dilate_out_grad].compute_inline()
+            In_grad = OP.output(0)
+            _schedule(Padded_out_grad, In_grad)
```

Comment:
raise error for unexpected cases, 

else:
    raise ValueError("cannot handle ...")

Reply:
It seems like its comparison with TF does not have unexpected cases yet. Will test more about it.

Reply:
The unexpected cases only happens when user mistakenly uses it, like call this schedule function with other declarations. 

Since you did not find a tag here, and not raise error for the else case this function can return safely with wrong results and user not knowing something is wrong

Reply:
Oh, I eventually got your point. Haha. Thank you!